rocky/tests/test_members.py

```diff
@@ -0,0 +1,162 @@
+    """
+    This will test if client can edit superuser at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), client_member.user)
+    with pytest.raises(PermissionDenied):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=superuser_member.organization.code, pk=superuser_member.id
+        )
+
+
+def test_redteamer_can_edit_superuser(rf, redteam_member, superuser_member, organization):
+    """
+    This will test if redteamer can edit superuser at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), redteam_member.user)
+    with pytest.raises(PermissionDenied):
+        OrganizationMemberEditView.as_view()(request, organization_code=organization.code, pk=superuser_member.id)
+
+
+def test_edit_superusers_from_different_organizations(rf, superuser_member, superuser_member_b):
+    """
+    This will test if a superuser from one organization can edit
+    a superuser from another organization at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), superuser_member.user)
+    # from OrganizationView
+    with pytest.raises(Http404):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=superuser_member_b.organization.code, pk=superuser_member_b.id
+        )
+
+
+def test_edit_admins_from_different_organizations(rf, admin_member, admin_member_b):
+    """
+    This will test if a admin from one organization can edit
+    a admin from another organization at the member edit view.
+    """
+
+    request = setup_request(rf.get("organization_member_edit"), admin_member.user)
+    # from OrganizationView
+    with pytest.raises(Http404):
+        OrganizationMemberEditView.as_view()(
+            request, organization_code=admin_member_b.organization.code, pk=admin_member_b.id
+        )
+
+
+def test_admin_edits_redteamer(rf, admin_member, redteam_member):
+    request = setup_request(
+        rf.post(
+            "organization_member_edit",
+            {"member_name": "Member name test", "status": "active", "trusted_clearance_level": 4},
+        ),
+        admin_member.user,
+    )
+    response = OrganizationMemberEditView.as_view()(
+        request, organization_code=redteam_member.organization.code, pk=redteam_member.id
+    )
+
+    assert response.status_code == 302
+    assert response.url == f"/en/{admin_member.organization.code}/"
+    resulted_request = setup_request(rf.get(response.url), admin_member.user)
+    resulted_response = OrganizationDetailView.as_view()(
+        resulted_request, organization_code=admin_member.organization.code
+    )
+    assert resulted_response.status_code == 200
+
+    # member list has been updated
+    assertContains(resulted_response, "Member name test")
+    assertContains(resulted_response, "Active")
+    assertContains(resulted_response, "Yes (L4)")
```

Comment:
Instead of calling the OrganizationDetailView and trying to parse the response, just check if it is saved in the database Something like

```
redteam_member.refresh_from_db()
assert redteam_member.trusted_clearance_level == 4
```

Reply:
This means the code that calls OrganizationDetailView can be removed from the test. This makes the test more efficient. Testing OrganizationDetailView itself is already done in other tests.